auto_submit/test/requests/check_pull_request_test.dart

```diff
@@ -325,15 +322,23 @@ void main() {
     });
 
     test('Merges only _kMergeCountPerRepo PR per cycle per repo', () async {
-      final PullRequest pr1 = generatePullRequest(prNumber: 22);
-      final PullRequest pr2 = generatePullRequest(prNumber: 23);
-      final PullRequest pr3 = generatePullRequest(prNumber: 24);
+      final PullRequest pr1 = generatePullRequest(prNumber: 22, repoName: 'flutter', login: 'flutter');
+      final PullRequest pr2 = generatePullRequest(prNumber: 23, repoName: 'flutter', login: 'flutter');
+      final PullRequest pr3 = generatePullRequest(prNumber: 24, repoName: 'flutter', login: 'cocoon');
       config = FakeConfig(githubService: githubService);
       checkPullRequest = CheckPullRequest(config: config, pubsub: pubsub);
-      Set shouldMergeSet = {pr1, pr2, pr3};
-      final Set mergeSet = await checkPullRequest.checkMerge(shouldMergeSet);
-      List<bool> mergeResult = await checkPullRequest.mergePullRequest(mergeSet);
-      expect(mergeSet.length, 2);
+      final Map<String, Set<PullRequest>> shouldMergeMap = <String, Set<PullRequest>>{};
+      Set<PullRequest> shouldMergeSet = {pr1, pr2, pr3};
+      for (PullRequest pr in shouldMergeSet) {
+        String repoName = pr.base!.repo!.slug().fullName;
+        if (!shouldMergeMap.containsKey(repoName)) {
+          shouldMergeMap[repoName] = <PullRequest>{};
+        }
+        shouldMergeMap[repoName]!.add(pr);
+      }
+
+      List<bool> mergeResult = await checkPullRequest.checkMerge(shouldMergeMap);
+      expect(mergeResult.length, 2);
```

Comment:
It would be useful to validate on the repo level: for each repo, we only merge 1 PR correctly.

Reply:
Sure Keyong, I added some response to show the repo info.